main.go

```diff
@@ -1,149 +1,174 @@
 
-// Main worker.
-func mainWorker(ctx context.Context, profile config.ImageProcessingProfile, result_chan chan<- error) {
-
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	// Get only file name from the path.
-	input_image_name := profile.GetAssignedFilePath()
-	input_image_name = filepath.Base(input_image_name)
-
-	log.Printf("[.] Processing image [%s] with profile [%s]\n", input_image_name, profile.ProfileName)
-
-	// Process image.
-	for {
-		select {
-		case <-ctx.Done(): // Check if context is cancelled.
-			log.Printf("[!] Context cancelled. Exiting...\n")
-			result_chan <- nil
-			return // Terminate goroutine.
-		default:
-			err := ProcessFile(profile) // Process image.
-			if err != nil {
-				log.Printf("[x] Error while processing image: %v\n", err)
-				result_chan <- err
-				return // Terminate goroutine.
+		// If directory not exists and create_dir is true, create directory.
+		if os.IsNotExist(err) {
+
+			if create_dir { // Making profile directory if not exists.
+				err = os.Mkdir(profile_dir, 0777)
+				if err != nil {
+					return "", err
+				}
+
+				// Writing default profile.
+				err = os.WriteFile(profile_file, []byte(config.GenerateDefaultConfig().ToYaml()), 0777)
+				if err != nil {
+					return "", err
+				}
+			} else { // If create_dir is false, return error.
+				return "", err
 			}
-			result_chan <- nil
-			return // Goroutine finished.
+		} else {
+			return "", err
 		}
 	}
+
+	return profile_file, err
 }
 
-// Main function.
+// Main function, defines arguments and flags.
 func main() {
 
 	// Context for main worker.
 	ctx := context.Background()
 
-	config_paths := new(inputConfigFilePaths)
-
-	// Parse command line.
-	flag.Var(config_paths, "f", "Config file path (can be specified multiple times)")
-	flag.Parse()
-
-	config_root := config.ProfileRoot{}
-
-	// Merge all config files, if any specified.
-	if len(*config_paths) != 0 {
-		loaded_configs := make([]config.ProfileRoot, 0) // Placeholder for loaded configs.
-		for _, path := range *config_paths {
-			conf, err := config.LoadConfigFromFile(path) // Load config file.
-			if err != nil {
-				log.Printf("[!] Error (%s) while loading config file: %s The config file will be ignored.\n", err, path)
+	app := &cli.App{
+		Name:  "Image Processing CLI",
+		Usage: "Batch process images",
+		Authors: []*cli.Author{
+			{
+				Name:  "h-alice",
+				Email: "admin@halice.art",
+			},
+			{
+				Name:  "natlee",
+				Email: "natlee.work@gmail.com",
+			},
+		},
+		Flags: []cli.Flag{
+			&cli.StringSliceFlag{
+				Name:  "f",
+				Usage: "Config file path (can be specified multiple times)",
+			},
+		},
+		Action: func(c *cli.Context) error {
+
+			// Placeholder for input config file paths.
+			loaded_configs := make([]config.ProfileRoot, 0) // Placeholder for loaded configs.
+
+			// Placeholder for config root.
+			config_root := config.ProfileRoot{}
+
+			// First, check if any argument is specified.
+			if c.NArg() == 0 {
```

Comment:
Parser is a considerable annoyance for many developers.

Reply:
So I think it's a good practice to make it an isolated file. : D